test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts

```diff
@@ -644,125 +652,117 @@ describe('CSOT spec prose tests', function () {
-       * 1. Using `client`, create a GridFS bucket (referred to as `bucket`) that wraps the `db` database.
-       * 1. Call `bucket.open_upload_stream()` with the filename `filename` to create an upload stream (referred to as `uploadStream`).
-       *    - Expect this to succeed and return a non-null stream.
-       * 1. Using `uploadStream`, upload a single `0x12` byte.
-       * 1. Call `uploadStream.close()` to flush the stream and insert chunks.
-       *    - Expect this to fail with a timeout error.
-       */
-      const failpoint: FailPoint = {
-        configureFailPoint: 'failCommand',
-        mode: { times: 1 },
-        data: {
-          failCommands: ['insert'],
-          blockConnection: true,
-          blockTimeMS: 150
-        }
-      };
-      await internalClient.db().admin().command(failpoint);
+      // 1. Using `internalClient`, drop and re-create the `db.fs.files` and `db.fs.chunks` collections.
+      // 2. Using `internalClient`, set the following fail point:
+      //     ```javascript
+      //     {
+      //         configureFailPoint: "failCommand",
+      //         mode: { times: 1 },
+      //         data: {
+      //             failCommands: ["insert"],
+      //             blockConnection: true,
+      //             blockTimeMS: 200
+      //         }
+      //     }
+      //     ```
+      // 3. Create a new MongoClient (referred to as `client`) with `timeoutMS=150`.
+      // 4. Using `client`, create a GridFS bucket (referred to as `bucket`) that wraps the `db` database.
+      // 5. Call `bucket.open_upload_stream()` with the filename `filename` to create an upload stream (referred to as
+      //     `uploadStream`).
+      //     - Expect this to succeed and return a non-null stream.
+      // 6. Using `uploadStream`, upload a single `0x12` byte.
+      // 7. Call `uploadStream.close()` to flush the stream and insert chunks.
+      //     - Expect this to fail with a timeout error.
+      await configureFailPoint(
+        this.configuration,
+        {
+          configureFailPoint: 'failCommand',
+          mode: { times: 1 },
+          data: {
+            failCommands: ['insert'],
+            blockConnection: true,
+            blockTimeMS: 200
+          }
+        },
+        this.configuration.url({ useMultipleMongoses: false })
+      );
 
       const bucket = new GridFSBucket(client.db('db'));
       const stream = bucket.openUploadStream('filename');
-      const data = Buffer.from('13', 'hex');
 
-      const fileStream = Readable.from(data);
-      const maybeError = await pipeline(fileStream, stream).then(
-        () => null,
-        error => error
+      const { result: maybeError, duration } = await measureDuration(() =>
+        pipeline(Readable.from(Buffer.from('13', 'hex')), stream).catch(error => error)
       );
       expect(maybeError).to.be.instanceof(MongoOperationTimeoutError);
+      expect(duration).to.be.within(150 - 15, 150 + 15);
```

Comment:
The spec tests don't discuss measuring duration and we discussed previously that our spec prose test implementations should be as identical to the spec as possible (ideally identical). Either we should remove this or move it to its own test, right?

Reply:
Will remove.